repository.go

```diff
@@ -534,6 +529,43 @@ func (r repository) update(cw contextWrapper, doc *Document, mutation Mutation,
 	return nil
 }
 
+func (r repository) applyMutates(cw contextWrapper, doc *Document, mutation Mutation, filter FilterQuery) error {
+	var (
+		baseQueries = []Querier{filter, mutation.Unscoped, mutation.Cascade}
+		queries     = baseQueries
+	)
+	if version, ok := r.lockVersion(*doc, mutation.Unscoped); ok {
+		Inc("lock_version").SkipReload().Apply(doc, &mutation)
+		queries = append(queries, LockVersion(version))
+	}
+
+	var (
+		pField string
+		query  = r.withDefaultScope(doc.data, Build(doc.Table(), queries...), false)
+	)
+
+	if len(doc.data.primaryField) == 1 {
+		pField = doc.PrimaryField()
+	}
+
+	if updatedCount, err := cw.adapter.Update(cw.ctx, query, pField, mutation.Mutates); err != nil {
+		return mutation.ErrorFunc.transform(err)
+	} else if updatedCount == 0 {
+		return NotFoundError{}
+	}
+
+	if mutation.Reload {
+		baseQuery := r.withDefaultScope(doc.data, Build(doc.Table(), baseQueries...), false)
+		if err := r.find(cw, doc, baseQuery.UsePrimary()); err != nil {
+			return err
+		}
+	} else if version, ok := r.lockVersion(*doc, mutation.Unscoped); ok {
+		doc.SetValue("lock_version", version+1)
+	}
+
```

Comment:
But abstracting away version_lock logic doesn't allow us to simply modify the version lock afterwards. 

Otherwise we could extend mutates and mark some of them to be "not included" in reload queries and to be computable without a reload. This would, for example, allow handling other increments with a version lock without reloading the document.